components/raftstore/src/store/fsm/store.rs

```diff
@@ -2393,6 +2413,41 @@ impl<'a, EK: KvEngine, ER: RaftEngine, T: Transport> StoreFsmDelegate<'a, EK, ER
             .collect();
         cb(regions);
     }
+
+    fn on_get_store_safe_ts(&self, key_range: KeyRange, cb: Box<dyn FnOnce(u64) + Send>) {
+        let meta = self.ctx.store_meta.lock().unwrap();
```

Comment:
If there are a lot of requests, this can become bottle neck easily.

Reply:
It won't be accessed per-request or per-transaction, instead, TiDB will get the store safe ts periodically in the interval of seconds.

Reply:
Can you add a comment explaining the context?

Reply:
Sure!